warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermOffsetPopulator.java

```diff
@@ -175,57 +148,43 @@ protected Range getRange(Set<Key> keys) {
             evaluationFilter.startNewDocument(docKey);
         }
         
+        String field;
+        String value;
+        TFKey tfKey = new TFKey();
         Map<String,TermFrequencyList> termOffsetMap = Maps.newHashMap();
         
-        while (tfSource.hasTop()) {
-            Key key = tfSource.getTopKey();
-            FieldValue fv = FieldValue.getFieldValue(key);
+        while (offsetIter.hasTop()) {
+            Key key = offsetIter.getTopKey();
+            tfKey.parse(key);
             
-            // add the zone and term to our internal document
-            Content attr = new Content(fv.getValue(), source.getTopKey(), evaluationFilter == null || evaluationFilter.keep(key));
+            field = tfKey.getField();
+            value = tfKey.getValue();
             
-            // no need to apply the evaluation filter here as the TermFrequencyIterator above is already doing more filtering than we can do here.
-            // So this filter is simply extraneous. However if the an EventDataQueryFilter implementation gets smarter somehow, then it can be added back in
-            // here.
-            // For example the AncestorQueryLogic may require this....
-            // if (evaluationFilter == null || evaluationFilter.apply(Maps.immutableEntry(key, StringUtils.EMPTY_STRING))) {
+            // Malformed keys will fail to parse, skip them.
+            if (field == null || value == null) {
+                log.error("Malformed term frequency key: " + key.toStringNoTime());
+                continue;
```

Comment:
I believe we need to call `offsetIter.next()` prior to `continue` in this case or we'll just get stuck evaluating the same key when an error is encountered.

Reply:
The TermFrequencyOffsetIterator only returns keys that match it's search space, so in theory it will never return a malformed key. A continue here *could* risk skipping a key, and thus a hit. Either way, an error will be logged for investigation. I think skipping a key is preferable to getting stuck in an infinite loop. Adding in a next call here.